src/gpu/intel/jit/codegen/codegen.cpp

```diff
@@ -831,9 +827,8 @@ class ir_to_ngen_t final : public codegen_extension_iface_t,
 template <typename ngen_generator_t>
 class expr_evaluator_t : public ir_visitor_t {
 public:
-    expr_evaluator_t(ngen_generator_t *host, const expr_binding_t &expr_binding,
-            ngen_register_scope_t &scope)
-        : host_(host), expr_binding_(expr_binding), scope_(scope) {}
+    expr_evaluator_t(ngen_generator_t *host, ngen_register_scope_t &scope)
+        : host_(host), expr_binding_(host->expr_binding()), scope_(scope) {}
```

Comment:
Like in the previous commit, does it make sense to use `host_->expr_binding()` directly instead of caching a pointer in the constructor?